File: FangZhiHu/app/src/main/java/network/exception/ResultManageFun.java
package network.exception;


import com.tongqinyuan.fangzhihu.api.ApiResponse;
import com.tongqinyuan.fangzhihu.common.LogUtil;

import io.reactivex.Flowable;
import io.reactivex.functions.Function;

/**
 * Created by yuantongqin on 2017/8/15.
 */

public class ResultManageFun<T> implements Function<Throwable, Flowable<ApiResponse<T>>> {
    private static final String TAG = ResultManageFun.class.getSimpleName();

    @Override
    public Flowable<ApiResponse<T>> apply(Throwable throwable) {
        LogUtil.e(TAG, "=跑出异常==" + throwable.getMessage());
        throwable.printStackTrace();
        return Flowable.error(ExceptionEngine.handleException(throwable));
    }

}

File: FangZhiHu/CommonLibKotlin/src/main/java/network/excepition/ResultManageFun.kt
package network.excepition

import io.reactivex.Flowable
import io.reactivex.functions.Function
import network.entity.ApiResponse

/**
 * Created by tongqinyuan on 2018/3/7.
 */
class ResultManageFun<T> :Function<Throwable,Flowable<ApiResponse<T>>> {
    override fun apply(t: Throwable): Flowable<ApiResponse<T>> {

        //log.e(TAG, "=跑出异常==" + throwable.getMessage());
        t.printStackTrace()
        return Flowable.error(ExceptionEngine.handleException(t))
    }
}